x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_exploration/components/regression_exploration/evaluate_panel.tsx

```diff
@@ -100,14 +148,85 @@ export const EvaluatePanel: FC<Props> = ({ jobConfig, jobStatus }) => {
       setTrainingEval({
         meanSquaredError: '',
         rSquared: '',
-        error: genErrorEval.error,
+        error: trainingErrorEval.error,
       });
     }
   };
 
+  const loadData = async ({ isTraining }: { isTraining?: { query: string; operator: string } }) => {
+    // searchBar query is filtering for testing data
+    if (isTraining !== undefined && isTraining.query === 'false') {
+      loadGeneralizationData();
+
+      const docsCountResp = await loadDocsCount({ isTraining: false });
+      if (docsCountResp.success === true) {
+        setGeneralizationDocsCount(docsCountResp.docsCount);
+      } else {
+        setGeneralizationDocsCount(null);
+      }
+
+      setTrainingDocsCount(0);
+      setTrainingEval({
+        meanSquaredError: '--',
+        rSquared: '--',
+        error: null,
+      });
+    } else if (isTraining !== undefined && isTraining.query === 'true') {
+      // searchBar query is filtering for training data
+      loadTrainingData();
+
+      const docsCountResp = await loadDocsCount({ isTraining: true });
+      if (docsCountResp.success === true) {
+        setTrainingDocsCount(docsCountResp.docsCount);
+      } else {
+        setTrainingDocsCount(null);
+      }
+
+      setGeneralizationDocsCount(0);
+      setGeneralizationEval({
+        meanSquaredError: '--',
+        rSquared: '--',
+        error: null,
+      });
+    } else {
+      // No is_training clause/filter from search bar so load both
+      loadGeneralizationData(false);
+      const genDocsCountResp = await loadDocsCount({
+        ignoreDefaultQuery: false,
+        isTraining: false,
+      });
+      if (genDocsCountResp.success === true) {
+        setGeneralizationDocsCount(genDocsCountResp.docsCount);
+      } else {
+        setGeneralizationDocsCount(null);
+      }
+
+      loadTrainingData(false);
+      const trainDocsCountResp = await loadDocsCount({
+        ignoreDefaultQuery: false,
+        isTraining: true,
+      });
+      if (trainDocsCountResp.success === true) {
+        setTrainingDocsCount(trainDocsCountResp.docsCount);
+      } else {
+        setTrainingDocsCount(null);
+      }
+    }
+  };
+
   useEffect(() => {
-    loadData();
-  }, []);
+    const hasIsTrainingClause =
+      searchQuery.bool &&
+      searchQuery.bool.must.filter(
+        (clause: any) => clause.match && clause.match[`${resultsField}.is_training`] !== undefined
+      );
+    const isTraining =
+      hasIsTrainingClause &&
+      hasIsTrainingClause[0] &&
+      hasIsTrainingClause[0].match[`${resultsField}.is_training`];
+
+    loadData({ isTraining });
+  }, [searchQuery]);
```

Comment:
In `results_table.tsx` we use `[JSON.stringify(searchQuery)])`, maybe we should do it here too? (Not sure if there's a chance we'd miss an update because `seachQuery` might be passed around by reference)

Reply:
Good call - updated in 5fc0e873a4f1f4c6dd9d1bf7d2b68dca439582a7